whistl/classifier.py

```diff
@@ -0,0 +1,251 @@
+
+
+if __name__ == '__main__':
+    # Load mapping file
+    # Load data
+    # Combine all dataframes into a single dataframe
+    # Convert the dataframe into a matrix
+    # Create the label vector based on the mapping file and matrix
+    # Create pytorch dataset from the matrix and labels
+    # train model on dataset
+
+    parser = argparse.ArgumentParser(description='This script trains a classifier to differentiate'
+                                                 ' between sepsis and healthy gene expresssion')
+    parser.add_argument('map_file',
+                        help='The label: sample mapping file created by label_samples.py')
+    parser.add_argument('data_dir',
+                        help='The directory containing gene expression data from refine.bio. '
+                             'This directory should contain only the results from unzipping the '
+                             'file downloaded from refine.bio.')
+    parser.add_argument('gene_file',
+                        help='A file containing a list of genes to be used in the analysis.')
+    parser.add_argument('--tune_study_count',
+                        help='The number of studies to put in the tuning set',
+                        default=2)
+    parser.add_argument('--seed', help='The random seed to use', default=42)
+    parser.add_argument('--num_epochs', help='The number of epochs to train the model',
+                        default=500)
+    parser.add_argument('--no_gpu', help='Use the CPU to train the network', action='store_false')
+    args = parser.parse_args()
+
+    device = None
+    if args.no_gpu:
+        device = torch.device('cpu')
+    else:
+        # We'll use the default GPU, this will need to change later if using multiple GPUs
+        device = torch.device('cuda')
+
+    # set random seeds
+    # https://pytorch.org/docs/stable/notes/randomness.html
+    np.random.seed(args.seed)
+    random.seed(args.seed)
+    torch.manual_seed(args.seed)
+    if torch.backends.cudnn.enabled:
+        torch.backends.cudnn.deterministic = True
+        torch.backends.cudnn.benchmark = False
+
+    sample_to_label = parse_map_file(args.map_file)
+
+    # Prepare data
+    train_dirs, tune_dirs = train_tune_split(args.data_dir, args.tune_study_count)
+    label_to_encoding = {'sepsis': 1, 'healthy': 0}
+
+    sys.stderr.write('Generating training dataset...\n')
+    train_dataset = dataset.ExpressionDataset(train_dirs, sample_to_label, label_to_encoding,
+                                              args.gene_file)
+    train_loader = DataLoader(train_dataset, batch_size=16, shuffle=True, num_workers=2,
+                              pin_memory=True)
+    sys.stderr.write('Generating tuning dataset...\n')
+    tune_dataset = dataset.ExpressionDataset(tune_dirs, sample_to_label, label_to_encoding,
+                                             args.gene_file)
+    tune_loader = DataLoader(tune_dataset, batch_size=16, shuffle=True, num_workers=2,
```

Comment:
Going off of my comment above about `DataLoaders`: out of curiosity, is there a reason to shuffle your tune data? It seems to me like it would be just another source of noise, but maybe I'm missing something.

It definitely helps to shuffle the training data (otherwise your model will overfit to the sample order, I've had this happen in the past) but I can't think of any reason you'd want to do it for the tuning set.

Reply:
You're right, I had just copied my `train_variable`, shuffling doesn't matter in the tune data